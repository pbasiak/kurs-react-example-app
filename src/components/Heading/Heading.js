import React from 'react';

import './Heading.scss';

export function Heading(props) {
    return (
        <h1 className="page-heading">
            {props.title}
        </h1>
    );
}
