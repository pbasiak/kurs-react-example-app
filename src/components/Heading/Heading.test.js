import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Heading } from './Heading';

jest.mock('./styled/PageHeading.js', () => {
    return {
        LargePageHeading: 'LargePageHeading',
    }
})

describe('<Heading />', () => {
    it('should render', () => {
        const wrapper = shallow(<Heading />);
        expect(wrapper);
    });

    it('should match snapshot', () => {
        const wrapper = shallow(<Heading/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should match snapshot with passed props', () => {
        const wrapper = shallow(<Heading title="test title" variant="secondary" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with prop variant', () => {
        const wrapper = shallow(<Heading variant='secondary' />);
        expect(wrapper.prop('variant')).toEqual('secondary');
    });

    it('should render with text as child passed with prop', () => {
        const wrapper = shallow(<Heading title='Example' />);
        expect(wrapper.text()).toEqual('Example');
    });
});