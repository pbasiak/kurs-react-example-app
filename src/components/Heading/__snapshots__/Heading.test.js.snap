// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `<LargePageHeading />`;

exports[`<Heading /> should match snapshot with passed props 1`] = `
<LargePageHeading
  variant="secondary"
>
  test title
</LargePageHeading>
`;
